File: src/main/java/utils/DBUtils.java
package utils;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.boot.MetadataSources;
import org.hibernate.boot.registry.StandardServiceRegistry;
import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
import org.hibernate.query.Query;

import java.util.ArrayList;
import java.util.List;

public class DBUtils {

    static final private StandardServiceRegistry registry = new StandardServiceRegistryBuilder().configure("hibernate_cfg.xml").build();
    static final private SessionFactory sessionFactory = new MetadataSources(DBUtils.registry).buildMetadata().buildSessionFactory();


    static public void insert(Object object) {
        Session session = sessionFactory.openSession();

        try (session) {
            Transaction tx = session.beginTransaction();
            session.persist(object);
            tx.commit();
        } catch (Exception e) {
            // log
        }
    }

    static public void saveOrUpdate(Object object) {
        Session session = sessionFactory.openSession();

        try (session) {
            Transaction tx = session.beginTransaction();
            session.saveOrUpdate(object);
            tx.commit();
        } catch (Exception e) {
            // log
        }
    }

    static public void delete(Object object) {
        Session session = sessionFactory.openSession();

        try (session) {
            Transaction tx = session.beginTransaction();
            session.delete(object);
            tx.commit();
        } catch (Exception e) {
            // log
        }
    }

    static public ArrayList selectAll(Class table) {
        Session session = sessionFactory.openSession();
        String hql = "FROM " + table.getName();
        Query query = session.createQuery(hql);
        List result = query.list();
        session.close();
        return (ArrayList) result;
    }

    static public ArrayList query(String sql) {
        Session session = sessionFactory.openSession();
        Query query = session.createSQLQuery(sql);
        List result = query.list();
        session.close();
        return (ArrayList) result;
    }

}

File: src/main/kotlin/com/github/sylux6/watanabot/utils/DBUtils.kt
package com.github.sylux6.watanabot.utils

import org.hibernate.boot.MetadataSources
import org.hibernate.boot.registry.StandardServiceRegistryBuilder

object DBUtils {

    private val registry = StandardServiceRegistryBuilder().configure("hibernate_cfg.xml").build()
    private val sessionFactory = MetadataSources(DBUtils.registry).buildMetadata().buildSessionFactory()


    fun insert(`object`: Any) {
        val session = sessionFactory.openSession()

        try {
            val tx = session.beginTransaction()
            session.persist(`object`)
            tx.commit()
        } catch (e: Exception) {
            // log
        }

    }

    fun saveOrUpdate(`object`: Any) {
        val session = sessionFactory.openSession()

        try {
            val tx = session.beginTransaction()
            session.saveOrUpdate(`object`)
            tx.commit()
        } catch (e: Exception) {
            // log
        }

    }

    fun delete(`object`: Any) {
        val session = sessionFactory.openSession()

        try {
            val tx = session.beginTransaction()
            session.delete(`object`)
            tx.commit()
        } catch (e: Exception) {
            // log
        }

    }

    fun selectAll(table: Class<*>): List<Any> {
        val session = sessionFactory.openSession()
        val hql = "FROM " + table.name
        val query = session.createQuery(hql)
        val result = query.resultList
        session.close()
        return result
    }

    fun query(sql: String): List<Any> {
        val session = sessionFactory.openSession()
        val query = session.createNativeQuery(sql)
        val result = query.resultList
        session.close()
        return result
    }

}
